packages/pigeon/lib/dart_generator.dart

```diff
@@ -67,44 +67,45 @@ String _getCodecName(Api api) => '_${api.name}Codec';
 ///
 /// class FooCodec extends StandardMessageCodec {...}
 void _writeCodec(Indent indent, String codecName, Api api, Root root) {
-  indent.write('class $codecName extends StandardMessageCodec ');
+  final Iterable<EnumeratedClass> codecClasses = getCodecClasses(api, root);
+  if (codecClasses.isEmpty) {
```

Comment:
nit:  I generally try to avoid this sort of construct where you explicitly say "_writeCodec()" but it doesn't actually write anything.  It makes the bit hard to reason about when what it really is is "_maybeWriteCodec()", then when you layer on layer after layer of "maybe" it gets kind of hard to reason about the code.  I would recommend pulling these if statements out at the callsite.  It isn't an official stance so feel free to ignore the suggestion if it is burdensome.

Reply:
This is a good point, initially I was still writing some lines in these functions. I removed that since and didn't consider relocating the logic.